internal/pkg/template/repository/manifest/records.go

```diff
@@ -156,3 +166,30 @@ func (v *TemplateRecord) DefaultVersionOrErr() (VersionRecord, error) {
 	}
 	return version, nil
 }
+
+func (v *TemplateRecord) HasComponent(components *model.ComponentsMap) bool {
+	for _, component := range v.Requirements.Components {
+		if _, found := components.Get(keboola.ComponentID(component)); found {
+			return true
+		}
+	}
+	return false
+}
+
+func (v *TemplateRecord) HasFeature(d keboola.FeaturesMap) bool {
+	for _, feature := range v.Requirements.Features {
+		if d.Has(feature) {
+			return true
+		}
```

Comment:
All features should be present :x: 

I suggest also naming change: `HasFeature` -> `CheckFeatures` (similarly elsewhere).